debezium-connector-mysql/src/test/java/io/debezium/connector/mysql/MySqlAntlrDdlParserTest.java

```diff
@@ -86,6 +86,102 @@ public void beforeEach() {
         properties.put("topic.prefix", "test");
     }
 
+    @Test
+    @FixFor("DBZ-6003")
+    public void shouldProcessCreateUniqueBeforePrimaryKeyDefinitions() {
+        String ddl = "CREATE TABLE `dbz_6003_1` (\n"
+                + "`id` INT ( 11 ) NOT NULL,\n"
+                + "`name` VARCHAR ( 255 ),\n"
+                + "UNIQUE KEY `test` ( `name` ),\n"
+                + "PRIMARY KEY ( `id` )\n"
+                + ") ENGINE = INNODB;";
+        parser.parse(ddl, tables);
+        assertThat(((MySqlAntlrDdlParser) parser).getParsingExceptionsFromWalker().size()).isEqualTo(0);
+        assertThat(tables.size()).isEqualTo(1);
+        Table table = tables.forTable(null, null, "dbz_6003_1");
+        assertThat(table.primaryKeyColumnNames().size()).isEqualTo(1);
+        assertThat(table.primaryKeyColumnNames().get(0)).isEqualTo("id");
+        assertThat(table.columnWithName("name").isOptional()).isTrue();
+
+        // add PRIMARY KEY and UNIQUE KEY
+        ddl = "create table dbz_6003_2 (id int not null, name varchar(255));"
+                + "alter table dbz_6003_2 add unique key dbz_6003_2_uk(name);"
+                + "alter table dbz_6003_2 add primary key(id);";
+        parser.parse(ddl, tables);
+        table = tables.forTable(null, null, "dbz_6003_2");
+        assertThat(table.primaryKeyColumnNames().get(0)).isEqualTo("id");
+        assertThat(table.columnWithName("name").isOptional()).isTrue();
+
+        // add UNIQUE KEY and PRIMARY KEY
+        ddl = "create table dbz_6003_3 (id int not null, name varchar(255));"
+                + "alter table dbz_6003_3 add (primary key id(id), unique key dbz_6003_3_uk(name));";
+        parser.parse(ddl, tables);
+        table = tables.forTable(null, null, "dbz_6003_3");
+        assertThat(table.primaryKeyColumnNames().get(0)).isEqualTo("id");
+        assertThat(table.columnWithName("name").isOptional()).isTrue();
+
+        // add UNIQUE KEY, drop PRIMARY KEY
+        ddl = "create table dbz_6003_4 (id int not null, name varchar(255), primary key(id));"
+                + "alter table dbz_6003_4 add unique key dbz_6003_4_uk(name);"
+                + "alter table dbz_6003_4 drop primary key;";
+        parser.parse(ddl, tables);
+        table = tables.forTable(null, null, "dbz_6003_4");
+        assertThat(table.primaryKeyColumnNames().size()).isEqualTo(0);
```

Comment:
@harveyyue  And here we get into an interesting problem, if you check the next case it is the same just different order of execution yet the result here is there is no key and in the following there is. General behaviour should be that if there is any change to either primary key or unique key then

- if PK exists it should be used
- if unique key exists it should be used